### PR TITLE
Datastore -- Comments/Contact feature

### PR DESCRIPTION
**Pull request by LindseyWeiskopf**
In this PR, I created a contact feature in which the user can fill out the form on the contact page, and their comment will be posted below the form and stored in persistent storage (through datastore). The doPost() function stores the comments into the datastore database by property while the doGet() function takes all of the entries from datastore and displays them on the contact pages using the formatting from the js file and the css file. The javascript file fetches the entries from the /comments extension and puts the entries into an unordered list, more specifically.

I'm really sorry if the commits are out of order for this PR--I did a lot of deleting and trying again, but hopefully the end files are more clean. Thank you!

http://lweiskopf-step-2020.appspot.com/contact.html

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -24,44 +31,51 @@
...
+    for (Entity entity : results.asIterable()) {
+      long id = entity.getKey().getId();
+      long timestamp = (long) entity.getProperty("timestamp");
+      String name = (String) entity.getProperty("name");
+      String email = (String) entity.getProperty("email");
+      String comment = (String) entity.getProperty("comment");
+      
+      Post post = new Post(id, name, email, comment, timestamp);
+      posts.add(post);
+    }
```
[the remark this answers is not in the document]
Comment (LindseyWeiskopf) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: Is it okay if I do this in my latest branch after I merge this one? I don't want to have a really messy merge conflict.

File: portfolio/src/main/java/com/google/sps/data/Post.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.google.sps.data;
+
+public final class Post {
```
[the remark this answers is not in the document]
Comment (LindseyWeiskopf) [reply portfolio/src/main/java/com/google/sps/data/Post.java:4]: Done!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (LindseyWeiskopf) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: I created more helper function in the addition-features pull request!

File: portfolio/src/main/java/com/google/sps/data/Post.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (shaquille-hall) [reply portfolio/src/main/java/com/google/sps/data/Post.java:4]: Prefer "//" for single line comments like this. Sorry if my previous comment was misleading.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -24,34 +31,32 @@
...
-    response.setContentType("text/html;");
-    response.getWriter().println(message);
+
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    
+    List<Post> posts = new ArrayList<>();
+
+    for (Entity entity : results.asIterable()) {
```
Comment (shaquille-hall) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: Moving this logic to a helper method may make this get easier to parse.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (shaquille-hall) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:55]: If that's the case, please add TODOs where relevant.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (LindseyWeiskopf) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:47]: yes - just added a todo!

File: portfolio/src/main/java/com/google/sps/data/Post.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (LindseyWeiskopf) [reply portfolio/src/main/java/com/google/sps/data/Post.java:4]: Sorry, that's my mistake -- I fixed them everywhere

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (LindseyWeiskopf) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:56]: Done.

**Pull request merged**